crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2688,6 +2688,57 @@ impl Backend {
 
         Ok(())
     }
+
+    /// Rollback the chain to a common height.
+    ///
+    /// The state of the chain is rewound using `rewind` to the common block, including the db,
+    /// storage, and env.
+    pub async fn rollback(&self, common_block: Block) -> Result<(), BlockchainError> {
```

Comment:
since this is basically same code as in `reorg` here

https://github.com/foundry-rs/foundry/blob/9f11e6df38f6b38173eba0cc1134ea14413a6ce8/crates/anvil/src/eth/backend/mem/mod.rs#L2634

can you please make a helper fn and use it in both cases?

Reply:
of course!
just made it so that `reorg` calls `rollback` before mining the new blocks 👌